src/services/userService.js

```diff
@@ -46,6 +46,32 @@ async function getUserByUsername(username) {
     return foundUser;
 }
 
+async function getUserById(userId) {
+    const result = await userDAO.getUserById(userId);
+    throwIfError(result);
+    const foundUser = result?.Item;
+    return foundUser;
+}
+
+async function updateUser(userId, requestBody) {
+    const foundUser = await getUserById(userId);
```

Comment:
You shouldn't have to make a database call if you use the auth middleware. res.locals.user should have all the user data.  Just have to make sure the user is the user they're updating and also pass the user object in.

Reply:
I just tried using res.locals.user but there's an issue when you do that. The user data you get from a token stays the same the entire time you're using that token. What I mean by that is if you make an update to one or more attributes of a user, it'll change in the database but you'll still be using the same user data before you made the update and not what's current in the database. You would have to create a new token to get the current user data and keep doing that after each update. I don't think we would want to do that so I think making a database call is necessary.

Reply:
good point! I didn't think about that. The token user wouldn't update unless they login again.